processor/src/main/java/org/mapstruct/ap/internal/model/MappingMethod.java

```diff
@@ -86,6 +86,10 @@ protected MappingMethod(Method method, List<LifecycleCallbackMethodReference> be
         this( method, method.getParameterNames(), beforeMappingReferences, afterMappingReferences );
     }
 
+    public void setParameters(List<Parameter> parameters) {
+        this.parameters = parameters;
```

Comment:
Please leave the class to be immutable and either add a dedicated construtor or convert the class to use a builder instead.